webapp/src/pages/ComponentsBookPage/ComponentsExamples/ToggleSwitchExample.tsx

```diff
@@ -0,0 +1,28 @@
+import { useState } from 'react';
+import {
+  ToggleSwitch,
+  Text,
+  TextColor,
+  TextVariant,
+  TextWeight,
+} from '../../../shared/components';
+import { BookSection } from '../BookSection';
+
+export const ToggleSwitchExample = () => {
+  const [isToggled, setIsToggled] = useState(false);
+  const onToggle = () => setIsToggled(!isToggled);
+  const value = isToggled ? 'toggled' : 'not toggled';
+
+  return (
+    <BookSection title="Chat Bubble component">
+      <ToggleSwitch isToggled={isToggled} onToggle={onToggle} />
+      <Text
+        variant={TextVariant.HEADING}
+        color={TextColor.LIGHT}
+        weight={TextWeight.BOLD}
+      >
+        {value}
+      </Text>
```

Comment:
suggestion: We could move the text label inside the component because this way, users could also click on the text to toggle the switch.

<details>

```diff
diff --git a/webapp/src/pages/ComponentsBookPage/ComponentsExamples/ToggleSwitchExample.tsx b/webapp/src/pages/ComponentsBookPage/ComponentsExamples/ToggleSwitchExample.tsx
index e8af7c5..776b280 100644
--- a/webapp/src/pages/ComponentsBookPage/ComponentsExamples/ToggleSwitchExample.tsx
+++ b/webapp/src/pages/ComponentsBookPage/ComponentsExamples/ToggleSwitchExample.tsx
@@ -1,28 +1,20 @@
 import { useState } from 'react';
-import {
-  ToggleSwitch,
-  Text,
-  TextColor,
-  TextVariant,
-  TextWeight,
-} from '../../../shared/components';
+import { ToggleSwitch } from '../../../shared/components';
 import { BookSection } from '../BookSection';
 
 export const ToggleSwitchExample = () => {
   const [isToggled, setIsToggled] = useState(false);
   const onToggle = () => setIsToggled(!isToggled);
-  const value = isToggled ? 'toggled' : 'not toggled';
 
   return (
-    <BookSection title="Chat Bubble component">
-      <ToggleSwitch isToggled={isToggled} onToggle={onToggle} />
-      <Text
-        variant={TextVariant.HEADING}
-        color={TextColor.LIGHT}
-        weight={TextWeight.BOLD}
-      >
-        {value}
-      </Text>
+    <BookSection title="Chat Bubble component" displayVertical>
+      <ToggleSwitch isToggled={isToggled} onToggle={onToggle} label="ReactJS" />
+      <ToggleSwitch isToggled={isToggled} onToggle={onToggle} label="NestJS" />
+      <ToggleSwitch
+        isToggled={isToggled}
+        onToggle={onToggle}
+        label=" I read and agree with the terms and conditions."
+      />
     </BookSection>
   );
 };
diff --git a/webapp/src/shared/components/ToggleSwitch/ToggleSwitch.css b/webapp/src/shared/components/ToggleSwitch/ToggleSwitch.css
index 51b517c..5afbf91 100644
--- a/webapp/src/shared/components/ToggleSwitch/ToggleSwitch.css
+++ b/webapp/src/shared/components/ToggleSwitch/ToggleSwitch.css
@@ -43,6 +43,7 @@ toggle-switch {
 
 .toggle-switch-text {
   position: relative;
-  left: 8rem;
-  top: -3.5rem;
+  left: 4rem;
+  top: -1rem;
+  white-space: nowrap;
 }
diff --git a/webapp/src/shared/components/ToggleSwitch/ToggleSwitch.tsx b/webapp/src/shared/components/ToggleSwitch/ToggleSwitch.tsx
index 6f87270..d638e60 100644
--- a/webapp/src/shared/components/ToggleSwitch/ToggleSwitch.tsx
+++ b/webapp/src/shared/components/ToggleSwitch/ToggleSwitch.tsx
@@ -1,19 +1,33 @@
+import Text, { TextColor, TextVariant, TextWeight } from '../Text/Text';
 import './ToggleSwitch.css';
 
 type ToggleSwitchProps = {
   isToggled: boolean;
   onToggle: React.ChangeEventHandler<HTMLInputElement>;
+  label?: string;
 };
 
 export default function ToggleSwitch({
   isToggled,
   onToggle,
+  label,
 }: ToggleSwitchProps) {
   return (
     <div className="toggle-switch">
       <label className="toggle-switch-label">
         <input type="checkbox" checked={isToggled} onChange={onToggle} />
         <span className="switch" />
+        {label && (
+          <div className="toggle-switch-text">
+            <Text
+              variant={TextVariant.PARAGRAPH}
+              color={TextColor.LIGHT}
+              weight={TextWeight.MEDIUM}
+            >
+              {label}
+            </Text>
+          </div>
+        )}
       </label>
     </div>
   );

```

</details>

Reply:
I think this is not exactly the label but a "dynamic" value to showcase toggle was working (you could replace it with any behaviour you wan't, for example make an api call)

Reply:
That being said, agree with adding an optional inside the component but I think is weird that the text could be clickable.

Usually you expect just the toggle button to be toggled, and being able to toggle it outside the switch would seem like a bug to me :sweat_smile: I see the label as something just informative

Reply:
Lets make our code accesible then 😄